viamonvif/discovery.go

```diff
@@ -274,31 +282,49 @@ func GetCameraInfo(
 	return cameraInfo, nil
 }
 
-// GetRTSPStreamURIsFromProfiles uses the ONVIF Media service to get the RTSP stream URLs for all available profiles.
-func GetRTSPStreamURIsFromProfiles(
+// GetMediaInfoFromProfiles uses the ONVIF Media service to get the RTSP stream URLs
+// and Snapshot URIs for all available profiles.
+func GetMediaInfoFromProfiles(
 	ctx context.Context,
 	dev OnvifDevice,
 	creds device.Credentials,
 	logger logging.Logger,
-) ([]string, error) {
+) ([]MediaInfo, error) {
 	resp, err := dev.GetProfiles(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	// Resultant slice of RTSP URIs
-	var rtspUris []string
-
-	// Iterate over all profiles and get the RTSP stream URI for each one
+	var mes []MediaInfo
+	// Iterate over all profiles and get the RTSP stream and snapshot URI for each one
 	for _, profile := range resp.Profiles {
-		uri, err := dev.GetStreamURI(ctx, profile, creds)
+		logger.Debugf("Looking up media info for profile: %s (token: %s)", profile.Name, profile.Token)
+		streamURI, err := dev.GetStreamURI(ctx, profile.Token, creds)
 		if err != nil {
 			logger.Warn(err.Error())
 			continue
 		}
 
-		rtspUris = append(rtspUris, uri.String())
+		snapshotURIString := ""
+		startTime := time.Now()
+		snapshotURI, err := dev.GetSnapshotURI(ctx, profile.Token, creds)
+		elapsed := time.Since(startTime)
+		logger.Infof("GetSnapshotURI took %s", elapsed)
+		if err != nil {
+			logger.Warn(err.Error())
+		} else {
+			snapshotURIString = snapshotURI.String()
+		}
+		if snapshotURIString == "" {
+			logger.Warnf("Snapshot URI is empty for profile %s: %s, adding media info with empty snapshot", profile.Name, streamURI.String())
+		}
```

Comment:
If and only if `err != nil` then `snapshotURIString == ""`

why do you have 2 if statements when you only need one?

the user doesn't need to know that we are adding media info with an empty snapshot

```suggestion
		if err != nil {
		        logger.Warnf("Unable to determine onvif snapshot URI  profile %s: %s, err: %s", profile.Name, streamURI.String(), err.Error())
		} else {
			snapshotURIString = snapshotURI.String()
		}
```